### PR TITLE
WRN-848: Updated to use `forwardCustom` and add `type` when forwarding custom events

### PR DESCRIPTION
**Pull request by MikyungKim**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
We've added `forwardCustom` in https://github.com/enactjs/enact/pull/2773.
And some of the custom events have no `type` when firing.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Fixed to use `forwardCustom` and add `type` when forwarding custom events.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-848

### Comments
Enact-DCO-1.0-Signed-off-by: Mikyung Kim (mikyung27.kim@lge.com)

### CODE REVIEW

File: packages/ui/Transition/Transition.js [hunk 1]
```diff
@@ -523,13 +519,13 @@ class Transition extends Component {
 	};
 
 	handleTransitionEnd = (ev) => {
-		forwardTransitionEnd(ev, this.props);
+		forwardCustom('onTransitionEnd')(ev, this.props);
```
Comment (0x64) [inline packages/ui/Transition/Transition.js:522]: It looks `ev` is meaningless.

Comment (0x64) [inline packages/ui/Transition/Transition.js:522]: What about replacing `ev` with `null` to avoid any misunderstanding later? :)

Comment (MikyungKim) [reply packages/ui/Transition/Transition.js:522]: 🆗

File: packages/ui/FloatingLayer/tests/FloatingLayer-specs.js [hunk 2]
```diff
@@ -30,4 +30,59 @@ describe('FloatingLayer Specs', () => {
...
+
+		const expected = 1;
+		const expectedType = {type: 'onOpen'};
+		const actual = handleOpen.mock.calls.length && handleOpen.mock.calls[0][0];
+
+		expect(handleOpen).toBeCalledTimes(expected);
+		expect(actual).toMatchObject(expectedType);
+	});
+
+	test('should fire onClose event with type when FloatingLayer is open', () => {
```
Comment (0x64) [inline packages/ui/FloatingLayer/tests/FloatingLayer-specs.js:51]: Please check the sentence.

File: packages/ui/FloatingLayer/tests/FloatingLayer-specs.js [hunk 3]
```diff
@@ -30,4 +30,59 @@ describe('FloatingLayer Specs', () => {
...
+			</Root>
+		);
+
+		const expectedType = {type: 'onClose'};
+		const actual = handleClose.mock.calls.length && handleClose.mock.calls[0][0];
+
+		expect(actual).toMatchObject(expectedType);
+	});
+
+	test('should fire onDimiss event with type when FloatingLayer is open', () => {
```
Comment (0x64) [inline packages/ui/FloatingLayer/tests/FloatingLayer-specs.js:72]: Please check the sentence.

File: packages/ui/RadioDecorator/tests/RadioDecorator-specs.js [hunk 4]
```diff
@@ -60,6 +60,54 @@ describe('RadioDecorator', () => {
...
+
+		userEvent.click(component);
+
+		const expected = 1;
+		const expectedType = {type: 'onClick'};
+		const actual = handleActivate.mock.calls.length && handleActivate.mock.calls[0][0];
+
+		expect(handleActivate).toBeCalledTimes(expected);
+		expect(actual).toMatchObject(expectedType);
+
```
Comment (0x64) [inline packages/ui/RadioDecorator/tests/RadioDecorator-specs.js:84]: Is this blank line intended?

File: packages/ui/RadioDecorator/tests/RadioDecorator-specs.js [hunk 5]
```diff
@@ -60,6 +60,54 @@ describe('RadioDecorator', () => {
...
+		userEvent.click(component);
+
+		const expected = 1;
+		const expectedType = {type: 'onClick'};
+		const actual = handleDeactivate.mock.calls.length && handleDeactivate.mock.calls[0][0];
+
+		expect(handleDeactivate).toBeCalledTimes(expected);
+		expect(actual).toMatchObject(expectedType);
+
```
Comment (0x64) [inline packages/ui/RadioDecorator/tests/RadioDecorator-specs.js:108]: Is this blank line intended?

File: packages/ui/Toggleable/tests/Toggleable-specs.js [hunk 6]
```diff
@@ -152,11 +152,14 @@ describe('Toggleable', () => {
 			data.onToggle();
 
 			const expected = 1;
+			const expectedType = {type: 'onToggle'};
```
Comment (0x64) [inline packages/ui/Toggleable/tests/Toggleable-specs.js:155]: What about adding a case for `toggleProp`? We already have `toggleProp: 'onJiggle'` test case in this file.

**Review by 0x64 — APPROVED** [review]

**Pull request merged**